fix: Shopping cart Exchange rate validation

marination opened the pull request:
Backport of https://github.com/frappe/erpnext/pull/27040

Shopping Cart Settings -> E Commerce Settings

marination committed:
fix: Shopping cart Exchange rate validation

- Use `get_exchange_rate` to check for price list exchange rate in cart settings
- Move cart exchange rate validation for Price List from hooks to doc event
- Call cart exchange rate validation on PL update only if PL is in cart and currency is changed

marination committed:
chore: Comment out obsolete test

- Modifying this test means considering extreme edge cases, which seems pointless now

nabinhait commented:
@marination We have changed the exchange rate API from `frankfurter` to `exchangerate.host` in version-13 as well. Do you need any changes due to that?

marination commented:
> @marination We have changed the exchange rate API from `frankfurter` to `exchangerate.host` in version-13 as well. Do you need any changes due to that?

@nabinhait this is in draft because v13 was still using frankfurter, this PR is fit for exchangerate.host

nabinhait committed:
Merge branch 'version-13-hotfix' into cart-pl-exchange-rate-hotfix

marination committed:
Merge branch 'version-13-hotfix' into cart-pl-exchange-rate-hotfix

marination committed:
fix: Remove snippet that got in due to cherry-pick from `develop`

- This snippet is not present in v13-hotfix. Via https://github.com/frappe/erpnext/pull/26520

marination commented:
Unrelated `test_appointment_cancel` failing
